src/EditorFeatures/Core.Wpf/Preview/LinesIndicator.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+using System.Windows.Shapes;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Text.Differencing;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Formatting;
+using Microsoft.VisualStudio.Text.Projection;
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Preview
+{
+    internal sealed class LinesIndicator
+    {
+        private readonly IAdornmentLayer _layer;
+        private readonly IWpfTextView _view;
+        private readonly IDifferenceViewer _viewer;
+        private readonly IClassificationFormatMap _classificationFormatMap;
+        private readonly IEditorFormatMap _editorFormatMap;
+        private readonly IClassificationType _lineNumberClassificaiton;
+        private readonly LinesIndicatorTextViewCreationListener _factory;
+
+        public LinesIndicator(IWpfTextView view, IDifferenceTextViewModel model, LinesIndicatorTextViewCreationListener factory)
+        {
+            _layer = view.GetAdornmentLayer(nameof(LinesIndicator));
+            _view = view;
+            _viewer = model.Viewer;
+            _factory = factory;
+            _classificationFormatMap = factory.ClassificationFormatMapService.GetClassificationFormatMap(view);
+            _editorFormatMap = factory.EditorFormatMapService.GetEditorFormatMap(view);
+            _lineNumberClassificaiton = factory.ClassificationTypeRegistryService.GetClassificationType("line number");
+
+            view.Closed += this.OnClosed;
+            view.LayoutChanged += this.OnLayoutChanged;
+        }
+
+        private void OnClosed(object sender, EventArgs e)
+        {
+            _view.Closed -= this.OnClosed;
+            _view.LayoutChanged -= this.OnLayoutChanged;
+        }
+
+        private void OnLayoutChanged(object sender, TextViewLayoutChangedEventArgs e)
+        {
+            foreach (var line in e.NewOrReformattedLines)
+            {
+                this.CreateVisuals(line);
+            }
+        }
+
+        private void CreateVisuals(ITextViewLine line)
+        {
+            if ((line.Extent.Length == PreviewFactoryService.Ellipsis.Length) && (line.Extent.GetText() == PreviewFactoryService.Ellipsis))
```

Comment:
Should this confirm that the ellipsis is on the combined buffer rather than actual text in one side of the diff?